client/src/components/workspaces/utils.ts

```diff
@@ -79,10 +86,42 @@ export async function workspaceShareClick(workspace: WorkspaceInfo, informationM
   await modal.onWillDismiss();
 }
 
+export async function workspaceFavoriteClick(
```

Comment:
This function could return the new list of favourite. This way the caller does not need to get it from the storage.